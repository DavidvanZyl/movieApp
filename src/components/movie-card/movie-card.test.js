import React from 'react';
import ReactDOM from 'react-dom';
import MovieCardComponent from './movie-card.component';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('MovieCard Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MovieCardComponent movie={{ title: 'Test' }} />);
  });

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
