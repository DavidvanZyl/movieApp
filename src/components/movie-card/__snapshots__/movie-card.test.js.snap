// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCardComponent
    movie={
      Object {
        "title": "Test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Card
        className="card"
        expandable={false}
        expanded={null}
        initiallyExpanded={false}
      >
        <CardMedia
          className="cardMedia"
          overlay={<CardTitle />}
        >
          <img
            alt="undefined Poster"
            className="bgImage"
          />
        </CardMedia>
      </Card>,
      "replace": false,
      "to": "/movie/undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <CardMedia
          className="cardMedia"
          overlay={<CardTitle />}
        >
          <img
            alt="undefined Poster"
            className="bgImage"
          />
        </CardMedia>,
        "className": "card",
        "expandable": false,
        "expanded": null,
        "initiallyExpanded": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            alt="undefined Poster"
            className="bgImage"
          />,
          "className": "cardMedia",
          "overlay": <CardTitle />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "undefined Poster",
            "className": "bgImage",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Card
          className="card"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}
        >
          <CardMedia
            className="cardMedia"
            overlay={<CardTitle />}
          >
            <img
              alt="undefined Poster"
              className="bgImage"
            />
          </CardMedia>
        </Card>,
        "replace": false,
        "to": "/movie/undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <CardMedia
            className="cardMedia"
            overlay={<CardTitle />}
          >
            <img
              alt="undefined Poster"
              className="bgImage"
            />
          </CardMedia>,
          "className": "card",
          "expandable": false,
          "expanded": null,
          "initiallyExpanded": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="undefined Poster"
              className="bgImage"
            />,
            "className": "cardMedia",
            "overlay": <CardTitle />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "undefined Poster",
              "className": "bgImage",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
